.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,92 @@
+name: Docker Image CI
+
+on:
+  push:
+    tags:
+     - v**
+  pull_request:
+    branches: [ master ]
+     
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
```

Comment:
I assume `packages` write permission is for publishing releases?

What is `contents` write permission used for?

Reply:
publishing releases, a setting on the project needs to be flipped as well.

Under Project settings, Actions, General:
![image](https://user-images.githubusercontent.com/4044356/230729435-96a544ab-5d9d-4720-a754-24e77319b7f1.png)

Reply:
So `packages` is for uploading the artefact, but `contents` is for creating a release object? It kind of looks like `contents` also includes complete write access to the whole repository, but I guess we can live with that?

Reply:
Yeah, not ideal. Fortunately, github doesn't allow any ol' random PR access to the GITHUB_TOKEN by default.